### DISCUSSION TITLE
Why is TinyMCE requiring an API key? Free plan limited to 1000 loads?

### DISCUSSION THREAD

**Discussion by rawcomposition** [tinymce/tinymce#9207] [General]
If TinyMCE is free, open source, and licensed under the MIT license, then why am I now getting threatening warnings sayin an API key is required? The pricing on the website for an API key indicates it's limited to 1000 loads. Is there a way to use TinyMCE without paying exorbitant prices?

**Comment by TheSpyder** [comment]
TinyMCE is (and always will be) available for free as an open source product. Our cloud service, however, is strictly limited - and that is what requires an API key.

There are plenty of free CDN options available, such as:
https://cdnjs.com/libraries/tinymce
https://www.jsdelivr.com/package/npm/tinymce
https://unpkg.com/browse/tinymce/

Or you can [download the community edition](https://www.tiny.cloud/get-tiny/) and host it yourself.

**Reply by rawcomposition** [reply]
Thanks for clarifying. I installed the npm package `@tinymce/tinymce-react`. I assumed this would be considered self hosted. Is that not the case?

**Reply by TheSpyder** [reply]
No, our integrations default to using our cloud service. They don't bundle TinyMCE (they are mostly agnostic to the TinyMCE version). There are a few options for including a self-hosted TinyMCE with the React integration:
https://www.tiny.cloud/docs/tinymce/6/react-pm-host/

At a high level, if the `tinymceScriptSrc` property isn't set and there is no `tinymce` global in the page it will load TinyMCE from our cloud service.
